src/command/subgraph/publish.rs

```diff
@@ -42,7 +52,49 @@ impl Publish {
         client_config: StudioClientConfig,
         git_context: GitContext,
     ) -> RoverResult<RoverOutput> {
+        // * if --allow-invalid-routing-url is not provided
+        // * AND a --routing-url is provided
+        // * AND the URL is unparsable
+        // we need to warn and prompt the user, else we can assume a publish
+        if !self.allow_invalid_routing_url {
+            if let Some(routing_url) = &self.routing_url {
+                if let Err(parse_error) = Url::parse(routing_url) {
+                    tracing::debug!("Parse error: {}", parse_error.to_string());
```

Comment:
```suggestion
                    tracing::debug!("URL parse error: {:?}", parse_error);
```

using the debug implementation here is a bit nicer for debug logs than `.to_string()` i think